src/components/text-editor/text-editor.tsx

```diff
@@ -0,0 +1,98 @@
+import { Component, Element, Prop, forceUpdate, h } from '@stencil/core';
+import { Editor, EditorProps } from '@toast-ui/react-editor';
+import { EditorType } from '@toast-ui/editor/types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import '@toast-ui/editor/dist/toastui-editor.css';
+
+/**
+ * The `limel-text-editor` component wraps the toast-ui react editor. This editor
+ * allows a rich text editing experience with markdown support.
+ */
+@Component({
+    tag: 'limel-text-editor',
+    shadow: true,
+    styleUrl: 'text-editor.scss',
+})
+export class TextEditor {
+    @Prop()
+    initialValue: string = '';
+
+    @Prop()
+    editType: EditorType = 'wysiwyg';
```

Comment:
Exposing external types like this can be dangerous, since if a breaking change is introduced we might not notice it. It's better to create our own type and map it to the original type